src/core/space_interaction_API.jl

```diff
@@ -285,8 +285,13 @@ function add_agent!(
     kwargs...,
 )
     id = nextid(model)
-    newagent = A(id, pos, properties...; kwargs...)
-    add_agent_pos!(newagent, model)
+    if any(isequal(:pos), fieldnames(A))
+        newagent = A(id, pos, properties...; kwargs...)
+        add_agent_pos!(newagent, model)
```

Comment:
Wait, isn't this a hit to performance? Please check if there is a more efficient way to test if a struct as a field `pos`.

Reply:
```suggestion
    if :pos in fieldnames(A)
        newagent = A(id, pos, properties...; kwargs...)
        add_agent_pos!(newagent, model)
```

I believe this is the idiomatic way of doing this.